### ISSUE TITLE
no error when providing wrong number of arguments for a decorated function

### ISSUE THREAD

**Issue by bashirmindee** [astral-sh/ty#2640]
### Summary

```python
# script.py
import functools

@functools.cache  # comment this line 
def func(arg1, arg2):
    return arg1 + arg2
    pass

print(func())
```
```bash
$ ty check script.py
All checks passed!
```
After commenting the decorator:
```bash
$ ty check script.py
error[missing-argument]: No arguments provided for required parameters `arg1`, `arg2` of function `func`
 --> script.py:8:7
  |
6 |     pass
7 |
8 | print(func())
  |       ^^^^^^
  |
info: Parameters declared here
 --> bla.py:4:9
  |
3 | # @functools.cache
4 | def func(arg1, arg2):
  |         ^^^^^^^^^^^^
5 |     return arg1 + arg2
6 |     pass
  |
info: rule `missing-argument` is enabled by default

Found 1 diagnostic
```

This is absolutely crucial because we use decorators extensively

**Comment by carljm** [comment]
Thanks for the report!

This is not a bug in ty. All other type checkers (pyright, mypy, pyrefly) give the same answer here. The problem is in typeshed; the [stub for `functools.cache`](https://github.com/python/typeshed/blob/main/stdlib/functools.pyi#L240) does not preserve the arguments of the decorated function, which is ultimately because the [stub for the returned `_lru_cache_wrapper`](https://github.com/python/typeshed/blob/main/stdlib/functools.pyi#L57) is only generic over its return type, not over a `ParamSpec`. 

The reasons for this typeshed bug are discussed extensively in https://github.com/python/typeshed/issues/11280

ty preserves the signature of decorated functions fine, when the decorator is appropriately annotated to preserve them.
